java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/MethodCallSymbolProvider.java

```diff
@@ -25,12 +33,50 @@ public List<SymbolInformation> get(SearchMatch match) {
         try {
             MethodReferenceMatch m = (MethodReferenceMatch) match;
             IMethod e = (IMethod) m.getElement();
-            SymbolInformation symbol = new SymbolInformation();
-            symbol.setName(e.getElementName());
-            symbol.setKind(convertSymbolKind(e));
-            symbol.setContainerName(e.getParent().getElementName());
-            symbol.setLocation(getLocation(e, match));
-            symbols.add(symbol);
+            if (this.query.contains(".")) {
+                ICompilationUnit unit = e.getCompilationUnit();
```

Comment:
Maybe it's worth checking if [the file has syntax errors](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fcore%2FICompilationUnit.html) before visiting all the nodes, as mentioned [here](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fcore%2FICompilationUnit.html)? Not sure if it will make a difference, since most probably any project to be analysed won't have syntax errors, but since this can be a bit costly it might be worth taking every precation

Reply:
@jmle trying to figure out what the cost is here. Honestly, if it isn't too bad in terms of total runtime, I think we should just make this by default as it improves accuracy for all cases.